txpipe/simulation.py

```diff
@@ -0,0 +1,544 @@
+from .base_stage import PipelineStage
+from .utils import choose_pixelization
+from .data_types import (
+    HDFFile,
+    ShearCatalog,
+    TextFile,
+    MapsFile,
+    FileCollection,
+    FiducialCosmology,
+    TomographyCatalog,
+    QPNOfZFile,
+)
+import glob
+import time
+import numpy as np
+
+
+class TXLogNormalGlass(PipelineStage):
+    """
+    Uses GLASS to generate a simulated catalog from lognormal fields
+    GLASS citation:
+    https://ui.adsabs.harvard.edu/abs/2023OJAp....6E..11T
+
+    Contamination is applied to the density field by poission sampling
+    the density field at a higher rate than target density, then
+    removing objects with prob proportional to 1/input_weight
+    This allows us to produce contaminated/ucontamined maps of the same
+    galaxy realisation
+    """
+
+    name = "TXLogNormalGlass"
+    parallel = False
+    inputs = [
+        ("mask", MapsFile),
+        ("lens_photoz_stack", QPNOfZFile),
+        ("fiducial_cosmology", FiducialCosmology),
+        ("input_lss_weight_maps", MapsFile),
+    ]
+
+    outputs = [
+        ("photometry_catalog", HDFFile),
+        ("lens_tomography_catalog_unweighted", TomographyCatalog),
```

Comment:
might be me that is dumb, but is this overwriting the photometry catalog or is it just adding to it/ making a new one?

Reply:
its making a new one. The idea is, this stage would be run at the start of the pipeline and its output gets used as the "photometry_catalog" (or just the ra,dec,z columns for now) for any later stages